lib/internal/util.js

```diff
@@ -320,14 +321,29 @@ const kCustomPromisifiedSymbol = SymbolFor('nodejs.util.promisify.custom');
 const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
 
 let validateFunction;
+let validateInteger;
 
-function promisify(original) {
+function promisify(original, options) {
   // Lazy-load to avoid a circular dependency.
-  if (validateFunction === undefined)
-    ({ validateFunction } = require('internal/validators'));
+  if (validateFunction === undefined || validateInteger === undefined) {
+    ({
+      validateFunction,
+      validateInteger,
+    } = require('internal/validators'));
+  }
 
   validateFunction(original, 'original');
 
+  // No validateObject so .map(util.promisify) can work
+  if (options == null || typeof options !== 'object') {
+    options = ObjectCreate(null);
+  }
```

Comment:
Ideally we would avoid creating a throw away object here.

Reply:
Global `NullObject` for that is WIP currently. :)